### PR TITLE
new gallery routes

### PR DESCRIPTION
**Pull request by Polyrhythm**
Adds routes for new galleries.

### CODE REVIEW

File: src/endpoints/gallery.js [hunk 1]
```diff
@@ -13,3 +13,52 @@ export default endpoint({
...
+    path: 'gallery/topic',
+    apiUrl: `${utils.API_URL}/${utils.API_VERSION}`,
+    get(topic, sort, page, window='all') {
+        const requestPath = `${this.path}/${topic}/${sort}/${window}/${page}`;
+        const options = utils.buildOptions(this.apiUrl, requestPath, 'get');
+
+        return this.imgurAPICall(options);
+    },
+    post: galleryPostEndpoint
+});
```
Comment (btipling) [inline src/endpoints/gallery.js]: How about:

```javascript
function galleryEndpoint(windowType, endPoint) {
  return {
     path: `gallery/${endPoint}`,
     apiUrl: `${utils.API_URL}/${utils.API_VERSION}`,
     get(topic, sort, page, window=windowType {
         const requestPath = `${this.path}/${topic}/${sort}/${window}/${page}`;
         const options = utils.buildOptions(this.apiUrl, requestPath, 'get');
 
         return this.imgurAPICall(options);
     },
     post: galleryPostEndpoint
 };
}
const WEEK = 'week';
const ALL = 'all';
export const subredditEndpoint = endpoint(galleryEndpoint(WEEK, 'r');
export const tagEndpoint = endpoint(galleryEndpoint(WEEK, 't');
export const searchEndpoint = endpoint(galleryEndpoint(ALL, 'search'));
export const topicEndpoint = endpoint(galleryEndpoint(ALL, 'topic'));
```

**Comment by btipling** [comment]
:star2:

**Comment by btipling** [comment]
walking around to get lunch I realized I should have suggested to put `endpoint` inside `galleryEndpoint`

**Comment by laposheureux** [comment]
:sheep:

**Pull request merged**